util/identity/defaults.go

```diff
@@ -22,3 +22,11 @@ import "fmt"
 func GetAzureIdentityName(clusterName, clusterNamespace, identityName string) string {
 	return fmt.Sprintf("%s-%s-%s", clusterName, clusterNamespace, identityName)
 }
+
+// GetASOSecretName formats the name of the ASO Secret created by the capz controller.
+func GetASOSecretName(clusterIdentitySecret, subscriptionID string) string {
+	if subscriptionID == "" {
```

Comment:
For my own understanding, when might subscription ID be empty here? It looks to me like that's not possible given this:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/3562a2a29df8efbb9adb43442dde748c890882a1/azure/scope/clients.go#L116-L121

Reply:
that's true, will remove the subscription check